sky/spot/controller.py

```diff
@@ -186,21 +196,19 @@ def run(self):
             # Kill the children processes launched by log_lib.run_with_log.
             subprocess_utils.kill_children_processes()
             spot_state.set_cancelled(self._job_id)
-        except exceptions.SpotJobFailedBeforeProvisionError as e:
-            # The exception will be caught when:
-            # None of the failovers are caused by resource unavailability;
-            # i.e., they are caused by errors before actual provisioning,
-            # e.g., InvalidClusterNameError, NotSupportedError,
-            # CloudUserIdentityError, etc.
-            logger.error(common_utils.format_exception(e.reason))
+        except exceptions.ProvisionPrechecksError as e:
+            # Please refer to the docstring of self._run for the cases when
+            # this exception can occur.
+            logger.error(common_utils.format_exception(e.reasons))
```

Comment:
Can this take a list? How about log the same str as L206-208?

Reply:
Great catch! Fixed. Thanks!